tools/rtpsrelay/GuidAddrSet.cpp

```diff
@@ -224,7 +224,7 @@ GuidAddrSet::record_activity(const AddrPort& remote_address,
 
   switch (drain_state_) {
   case DrainState::DS_NORMAL:
-    if (!addr_set_stats.allow_stun_responses) {
+    if (!addr_set_stats.allow_stun_responses && !addr_set_stats.in_denied_partition) {
```

Comment:
I'm not doubting this is correct, but can someone explain the logic here?

What I'm following so far is that if we're not draining (DrainState::DS_NORMAL) and we see some activity from an address (src_guid -> cass -> addr_set_stats), then we would normally toggle on `allow_stun_responses` decrement `mark_count_`, but now we don't do that if the address was in a denied partition.

What are `mark_count_` and `mark_budget_`?